pkg/serviceaccounts/controllers/pull_secrets.go

```diff
@@ -8,14 +8,16 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/client-go/kubernetes"
-	api "k8s.io/kubernetes/pkg/apis/core"
 )
 
 // findDockercfgSecret checks all the secrets in the namespace to see if the token secret has any existing dockercfg secrets that reference it
 func findDockercfgSecrets(client kubernetes.Interface, tokenSecret *corev1.Secret) ([]*corev1.Secret, error) {
 	dockercfgSecrets := []*corev1.Secret{}
 
-	options := metav1.ListOptions{FieldSelector: fields.OneTermEqualSelector(api.SecretTypeField, string(v1.SecretTypeDockercfg)).String()}
+	// Field constants were removed in the v1.21 API with no immediate replacement.
+	// Moving to the external API was marked a TODO, but never implemented.
+	// See https://github.com/kubernetes/kubernetes/pull/90105
+	options := metav1.ListOptions{FieldSelector: fields.OneTermEqualSelector("type", string(v1.SecretTypeDockercfg)).String()}
```

Comment:
/cc @damemi 

You lgtm-ed this PR upstream - looks like we don't have an api replacement here?

Reply:
Sorry about that, in the discussion upstream it was decided that some of these field constants should actually not be in the external API, since the original comments indicated they were specific to the internal api. `SecretTypeField` was one of them, which I'm not sure if you saw the discussion on it but it may give some more relevant insight: https://github.com/kubernetes/kubernetes/pull/90105#discussion_r529890461